snapcraft.yaml

```diff
@@ -195,6 +198,31 @@ parts:
       echo "Creating inspect hook"
       cp $KUBE_SNAP_ROOT/scripts/inspect.sh .
 
+      # create bash completion script for microk8s.docker, microk8s.kubectl and microk8s.istio
+      # create docker completion. At this stage, the latest from the docker repository is used as a basis, it should
+      # probably be replaced with something else which would bring the completion matching the specific version inside
+      # microk8s
+
+      curl -L https://raw.githubusercontent.com/docker/cli/master/contrib/completion/bash/docker | sed "s/complete -F _docker docker docker.exe dockerd dockerd.exe/complete -F _docker microk8s.docker/g" > docker.bash
```

Comment:
We should pin the docker.io package we pull from the deb repository https://github.com/ubuntu/microk8s/blob/master/snapcraft.yaml#L93 . The reason is that for GPU support we have to cherry-pick the correct nvidia debs (eg https://github.com/ubuntu/microk8s/blob/master/snapcraft.yaml#L223). Pinning the docker.io package would allow us to find the exact bash completion script.

Reply:
I agree (I figured we could do it later on). 

In any case, is there a common path to where the docker package is installed in both cases? When installing from the regular docker.io package, the file is located in parts/docker/install/usr/share/bash-completion/completions/docker. I am not sure where it would be located in the nvidia case though.